controllers/apps/components/replication_set_utils.go

```diff
@@ -139,30 +124,16 @@ func HandleReplicationSetRoleChangeEvent(cli client.Client,
 		return nil
 	}
 
-	oldPrimaryPod, err := getAndCheckReplicationSetPrimaryPod(reqCtx.Ctx, cli, *cluster, compName)
-	if err != nil {
-		reqCtx.Log.Info("handleReplicationSetRoleChangeEvent gets old primary pod failed", "error", err)
-		return err
-	}
-	if oldPrimaryPod == nil {
-		return nil
-	}
-	// pod is old primary and newRole is secondary, it means that the old primary needs to be changed to secondary,
-	// we do not deal with this situation here, the demote labeling process of old primary to secondary is handled
-	// in another reconciliation triggered by role change event from secondary -> new primary,
-	// this is to avoid simultaneous occurrence of two primary or no primary at the same time
-	if oldPrimaryPod.Name == pod.Name {
-		reqCtx.Log.Info("pod is old primary and new role is secondary, do not deal with this situation",
-			"podName", pod.Name, "newRole", newRole)
+	// pod is old primary and newRole is secondary, it means that the old primary needs to be changed to secondary
+	if newRole == constant.Secondary {
```

Comment:
This will be more concise:

if err := updateObjRoleLabel(reqCtx.Ctx, cli, *pod, newRole); err != nil {
    return err
}
reqCtx.Log.Info("succeed to update pod role label, pod: %s, role: %s", pod.Name, newRole)